src/Client/Pagination/PaginationSession.ts

```diff
@@ -0,0 +1,14 @@
+// SPDX-FileCopyrightText: 2024 - 2025 Gnuxie <Gnuxie@protonmail.com>
+//
+// SPDX-License-Identifier: AFL-3.0
+
+import { Result } from '@gnuxie/typescript-result';
+import { PaginationChunk } from './PaginationChunk';
+
+/**
+ * Stateful pagination session that fascilitates iteration.
+ */
+export interface PaginationSession<ChunkItem> {
+  /** Fetch the next page (based on direction set at creation). */
+  nextPage(): Promise<Result<PaginationChunk<ChunkItem> | null>>;
+}
```

Comment:
`PaginationSession` isn't used for anything.

Reply:
we'll leave it just in case we need to do something crazy later and we are forget how anything works.